siac/main.go

```diff
@@ -93,6 +106,18 @@ func apiPost(call, vals string) (*http.Response, error) {
 		return nil, errors.New("no response from daemon")
 	}
 	// check error code
+	if resp.StatusCode == http.StatusUnauthorized {
+		resp.Body.Close()
+		// Prompt for password and retry request with authentication.
+		password, err := speakeasy.Ask("API password: ")
+		if err != nil {
+			return nil, err
+		}
+		resp, err = api.HttpPOSTAuthenticated("http://"+addr+call, vals, password)
+		if err != nil {
+			return nil, errors.New("no response from daemon - authentication failed")
+		}
+	}
```

Comment:
hmm, if `siac` is going to be asking for a password every time you run it, I can see why you would want to leave the information api calls exposed.